### PR TITLE
Refactor makefiles and do some cleaning

### PR DESCRIPTION
**Pull request by cscetbon**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -221,16 +225,9 @@ jobs:
...
-          -Dsonar.login=${SONAR_LOGIN} \
           -Dsonar.log.level=INFO
           cat -n .scannerwork/report-task.txt
 
-    - save_cache:
-        name: Cache SonarQube Scanner
-        key: sonar-scanner-3.3.0.1492-linux
-        paths:
-          - sonar-scanner-3.3.0.1492-linux
-          - /home/circleci/cassandra-k8s-operator/.sonar
```
Comment (allamand) [inline .circleci/config.yml]: Why move save cache before action ??

File: .circleci/config.yml [hunk 2]
```diff
@@ -134,7 +133,6 @@ jobs:
         paths:
           - build/_output
           - pkg/apis/db/v1alpha1/zz_generated.deepcopy.go
-          - pkg/apis/db/v1alpha1/zz_generated.deepcopy.go
```
Comment (allamand) [inline .circleci/config.yml]: We need to cache the other generated file

File: docker/circleci/Dockerfile [hunk 3]
```diff
@@ -103,12 +87,6 @@ RUN mkdir -p $GOPATH/src/github.com/operator-framework \
...
     && rm -rf /go/pkg/dep/sources/
 
-
-
-# END IMAGE CUSTOMIZATIONS
-
-#USER circleci
-
-WORKDIR cassandra-k8s-operator
+WORKDIR /go/cassandra-k8s-operator
```
Comment (allamand) [inline docker/circleci/Dockerfile]: Why adding /go there ?

Comment (cscetbon) [reply docker/circleci/Dockerfile]: `Follow best practices`

File: .circleci/config.yml (hunk 1 above)
Comment (cscetbon) [reply .circleci/config.yml]: just not needed later

**Comment by cscetbon** [comment]
> Are you sure about this ?

Blocked due to free-plan-docker-layer-caching-unavailable

File: docker/circleci/Dockerfile (hunk 3 above)
Comment (allamand) [reply docker/circleci/Dockerfile]: There is no need since we used go modules

Comment (cscetbon) [reply docker/circleci/Dockerfile]: Not sure we're on the same page. What do you mean ? I'm just saying that it's a best practice to use absolute paths as workdir values, see https://docs.docker.com/v17.09/engine/userguide/eng-image/dockerfile_best-practices/

Comment (allamand) [reply docker/circleci/Dockerfile]: Just why adding a /go in the path ?

File: .circleci/config.yml (hunk 2 above)
Comment (allamand) [reply .circleci/config.yml]: Why ?

Comment (cscetbon) [reply .circleci/config.yml]: We don’t need to cache the same file twice and there is only one generated file. Default is not there anymore

File: docker/circleci/Dockerfile (hunk 3 above)
Comment (cscetbon) [reply docker/circleci/Dockerfile]: Read the part on workdir, it must be absolute. And this is where it goes when we don’t specify it. So I’m not changing anything, just applying best practices

**Pull request merged**